sys/include/stdio_uart.h

```diff
@@ -19,6 +19,27 @@
  * USEMODULE += stdin
  * ```
  *
+ * @attention   Using STDIO over UART from interrupt context is not officially
+ *              supported
+ *
+ * For testing purposes and using STDIO within an ISR should mostly work good
+ * enough and for some platforms even reliable. Production code however should
+ * fully avoid any access to STDIO from interrupt context. Instead, e.g. an
+ * event could be posted to an @ref sys_event and the actual STDIO operation
+ * being deferred to thread context.
+ *
+ * Some reasons why RIOT does not officially support STDIO over UART from ISR:
+ * 1. UART is *slow* and the system easily remains in interrupt context for
+ *    unacceptable long time.
+ * 2. An UART implementation might choose to not guard an atomic sequence of
+ *    interactions with the hardware by disabling IRQs but using a mutex, e.g.
+ *    because the sequence takes long. Locking a mutex from interrupt
+ *    context will result in a deadlock, if the mutex is already locked.
```

Comment:
This argument becomes invalid by supporting stdio from UART; an implementation must not use a mutex for synchronization in order to allow for stdio from UART for debugging. I will drop it.

Reply:
Hmm, a list of two arguments doesn't look good. Now I have to come up with one more good argument :-D